src/pkgcheck/checks/codingstyle.py

```diff
@@ -903,3 +903,75 @@ def feed(self, pkg):
                         yield RedundantDodir(
                             cmd.group('cmd'), line=dodir.group('call'),
                             lineno=lineno - 1, pkg=pkg)
+
+
+class UnquotedVariable(results.LineResult, results.Style):
+    """Variable is unquoted, but should be quoted"""
+
+    def __init__(self, variable, **kwargs):
+        super().__init__(**kwargs)
+        self.variable = variable
+
+    @property
+    def desc(self):
+        return f"{self.lineno}: Variable {self.variable} should be quoted"
+
+
+class UnquotedVariableCheck(Check):
+    """Scan ebuild for variables that should be quoted like D, FILESDIR, etc."""
+
+    _source = sources.EbuildParseRepoSource
+    known_results = frozenset([UnquotedVariable])
+
+    message_commands = [
+        "die", "echo", "eerror", "einfo", "elog", "eqawarn", "ewarn"
+    ]
+    var_names = [
+        "D", "DISTDIR", "FILESDIR", "S", "T", "ROOT", "BROOT", "WORKDIR",
+        # EAPI=3/Prefix vars
+        "ED", "EPREFIX", "EROOT",
+	    # variables for games.eclass
+		"Ddir", "GAMES_PREFIX_OPT", "GAMES_DATADIR",
+		"GAMES_DATADIR_BASE", "GAMES_SYSCONFDIR", "GAMES_STATEDIR",
+		"GAMES_LOGDIR", "GAMES_BINDIR",
+	    # variables for multibuild.eclass
+	    "BUILD_DIR",
+    ]
+
+    def _var_needs_quotes(self, pkg, node):
+        pnode = node.parent
+        while pnode != node:
+            if pnode.type == 'string':
+                return False
+            elif pnode.type == 'variable_assignment':
+                # Variable is part of a simple shell assignment, it does not
+                # need to be quoted. Ie something like:
+                #
+                # S=${WORKDIR}/${PN}
+                #
+                # is OK.
```

Comment:
Maybe compress this comment a little? Your decision.

```suggestion
                # Variable is part of a simple shell assignment, it does not
                # need to be quoted. Ie something like: S=${WORKDIR}/${PN} is OK.
```